src/TranslatesFactoryData.php

```diff
@@ -11,16 +11,30 @@ private static function isFactory($item): bool
         return $item instanceof BaseFactory || $item instanceof FactoryBuilder;
     }
 
-    private function transformFactoriesToRelationIds(array $defaultModelFields): array
+    private static function isCallable($field): bool
     {
-        return collect($defaultModelFields)
-            ->map(function ($item) {
-                if ($this->isFactory($item)) {
-                    return $item->create()->getKey();
-                }
-
-                return $item;
-            })
-            ->toArray();
+        return is_callable($field) && ! is_string($field) && ! is_array($field);
+    }
+
+    private function transformModelFields(array $defaultModelFields): array
```

Comment:
I also renamed this because i felt it wasn't just transforming factories anymore, but not super opinionated on this, so if you feel strongly about the name, i can change it back.